lib/pbench/server/cache_manager.py

```diff
@@ -194,6 +181,82 @@ def make_cache_object(dir_path: Path, path: Path) -> CacheObject:
     )
 
 
+class LockRef:
+    """Keep track of a cache lock passed off to a caller"""
+
+    def __init__(self, lock: Path, exclusive: bool = False, wait: bool = True):
+        """Initialize a lock reference
+
+        The lock file is opened in "w+" mode, which is "write update": unlike
+        "r+", this creates the file if it doesn't already exist, but still
+        allows lock conversion between LOCK_EX and LOCK_SH.
+
+        Args:
+            lock: the path of a lock file
+            exclusive: lock for exclusive access
+            wait: wait for lock
+        """
+        self.lock = lock.open("w+")
+        self.locked = False
+        self.exclusive = exclusive
+        self.unlock = True
+        self.wait = wait
+
+    def acquire(self) -> "LockRef":
+        """Acquire the lock"""
+
+        cmd = fcntl.LOCK_EX if self.exclusive else fcntl.LOCK_SH
+        if not self.wait:
+            cmd |= fcntl.LOCK_NB
+        fcntl.lockf(self.lock, cmd)
+        self.locked = True
+        return self
+
+    def release(self):
+        """Release the lock and close the lock file"""
+
+        exception = None
+        try:
+            fcntl.lockf(self.lock, fcntl.LOCK_UN)
+            self.lock.close()
+            self.locked = False
+            self.exclusive = False
+        except Exception as e:
+            exception = e
+        finally:
+            # Release our reference to the lock file so that the
+            # object can be reclaimed.
+            self.lock = None
```

Comment:
This is...odd.  There are two problems:  the first is that it means that a `LockRef` object cannot be reused after the lock is released, which seems...unsatisfying (and unnecessary, and possibly surprising); the second is that (I think) this would be handled implicitly by Python, and we shouldn't need to clutter our code with explicit clean-up operations.

We should be able to delegate the close operation and the release of the file object to the destruction of the `LockRef` instance.  This would enable the caller to reacquire the lock using the same `LockRef` instance.  If you're really concerned about the timely release of the underlying file, then that should be encapsulated in a separate method which the caller can call if appropriate (and, otherwise, Python will handle it) -- it shouldn't be spread across various parts of the `release()` operation.